tests/test_workers/test_tasks/test_gomod.py

```diff
@@ -7,12 +7,49 @@
 from cachito.workers import tasks
 
 
+# TODO: test multiple valid go packages
+# TODO: test package at root, one at non-root
 @pytest.mark.parametrize(
-    "dep_replacements, expect_state_update",
+    "dep_replacements, expect_state_update, pkg_config, pkg_results",
     (
-        (None, True),
-        (None, False),
-        (False, [{"name": "github.com/pkg/errors", "type": "gomod", "version": "v0.8.1"}]),
+        (
+            None,
+            True,
+            None,
+            {"present": {".": True}, "relpath": {".": "./go.mod"}, "sourcedir": {".": "./"}},
+        ),
+        (
+            None,
+            False,
+            None,
+            {"present": {".": True}, "relpath": {".": "./go.mod"}, "sourcedir": {".": "./"}},
+        ),
+        (
+            False,
+            [{"name": "github.com/pkg/errors", "type": "gomod", "version": "v0.8.1"}],
+            None,
+            {"present": {".": True}, "relpath": {".": "./go.mod"}, "sourcedir": {".": "./"}},
+        ),
+        (
+            None,
+            False,
+            [{"path": "bar"}, {"path": "foo"}],
+            {
+                "present": {"bar": True, "foo": True},
+                "relpath": {"bar": "./bar/go.mod", "foo": "./foo/go.mod"},
+                "sourcedir": {"bar": "./bar/", "foo": "./foo/"},
+            },
+        ),
+        (
+            None,
+            False,
+            [{"path": "."}, {"path": "foo"}],
+            {
+                "present": {".": True, "foo": True},
+                "relpath": {".": "./go.mod", "foo": "./foo/go.mod"},
+                "sourcedir": {".": "./", "foo": "./foo/"},
+            },
```

Comment:
Honestly not sure what is the original purpose of the `expect_state_update` param, but could we set it to True for these scenarios? It seems that some useful checks are skipped when it's False

Reply:
You are right. I think I meant to go back and do this... but forgot. It should be resolved now.